activiti-core/activiti-image-generator/src/main/java/org/activiti/image/impl/DefaultProcessDiagramCanvas.java

```diff
@@ -1576,27 +1580,53 @@ public void drawMultiInstanceMarker(boolean sequential,
     }
 
     public void drawHighLight(int x,
-                              int y,
-                              int width,
-                              int height) {
+        int y,
+        int width,
+        int height,
+        boolean current) {
         Paint originalPaint = g.getPaint();
         Stroke originalStroke = g.getStroke();
 
-        g.setPaint(HIGHLIGHT_COLOR);
+        if (current) {
+            g.setPaint(HIGHLIGHT_CURRENT_COLOR);
+        } else {
+            g.setPaint(HIGHLIGHT_COMPLETED_ACTIVITY_COLOR);
+        }
```

Comment:
You could pass directly the color as a parameter instead of passing boolean and doing `if`, `else`.
If you don't want to expose to colors constant outside of this class you can create another two other dedicated methods that will call the current one passing the right color.

Reply:
Makes sense